File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    answer = ''
    x = 0
    i = 0
    
    for a in polynomial.split(' + '):
        if 'x' in a:
            if len(a) == 1:
                x += 1
            else:
                x += int(a[:len(a)-1])
        else:
            i += int(a)
    
    if x > 1 and i > 1:
        answer = str(x) + 'x + ' + str(i)
    elif x > 1 and i == 0:
        answer = str(x) + 'x'
    elif x == 1 and i != 0:
        answer = 'x + ' + str(i)
    elif x == 1:
        answer = 'x'
    elif i != 0:
        answer = str(i)
    
    return answer

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.java
import java.util.Arrays;

class Solution {
    public String solution(String polynomial) {
        String answer = "";
        int x = 0;
        int i = 0;
        
        for (String p : polynomial.split(" [+] ")) {
            if (p.charAt(p.length()-1) == 'x') {
                if (p.length() == 1) {
                    x++;
                } else {
                    x += Integer.parseInt(p.substring(0, p.length()-1));    
                }
            } else {
                i += Integer.parseInt(p);
            }
        }
        
        if (x > 1 && i > 0) {
            answer = x + "x + " + i;
        } else if ( x == 1 && i > 0) {
            answer = "x + " + i;
        } else if ( x == 1 && i == 0) {
            answer = "x";
        } else if ( x == 0) {
            answer = i + "";       
        } else if ( i == 0 ) {
            answer = x + "x";
        }
            
        return answer;
    }
}